src/ripple/protocol/impl/STObject.cpp

```diff
@@ -99,9 +99,19 @@ void STObject::set (const SOTemplate& type)
     }
 }
 
-bool STObject::setType (const SOTemplate& type)
+std::string STObject::setType (const SOTemplate& type)
 {
-    bool valid = true;
+    std::string result;
+
+    auto reportErr = [] (std::string const& field, char const* description)
+    {
+        std::stringstream ss;
+        ss << "Field '" << field << "' " << description;
+        std::string text {ss.str()};
+        JLOG (debugLog().error()) << "STObject::setType failed: " << text;
+        return text;
+    };
```

Comment:
Since the call sites assign `result`, you might consider...
```
    auto reportErr = [&result] (
        std::string const& field, char const* description)
    {
        result = "STObject::setType failed: Field '"
            + field + "' " + description;
        JLOG(debugLog().error()) << result;
    };
```

Reply:
Hi @miguelportilla.  Thanks for looking.  I thought about that, but felt the semantics were clearer by returning the string.

Regardless, I'm working on a significant modification to the pull request so `setType()` throws.  You may want to hold off further efforts not he review until I've finished that modification.

Sorry for the extra work...